File: inJava/app/src/main/java/com/barrygithub/rickandmortyapp/data/Repository.java
package com.barrygithub.rickandmortyapp.data;

import com.barrygithub.rickandmortyapp.data.LocalDatasource.entities.Character;
import com.barrygithub.rickandmortyapp.data.LocalDatasource.entities.EntityDb;
import com.barrygithub.rickandmortyapp.data.LocalDatasource.entities.MetaData;

import java.util.List;

import io.reactivex.rxjava3.core.Single;

/**
 * Project RickAndMortyApp
 * Created by Barry Zea H. on 3/04/23.
 * Copyright (c) Barry Zea H. All rights reserved.
 **/
public interface Repository {

    Single<String> getDataFromApiGraphql(int page);
    void saveMetadataInDb(MetaData metadata);
    void saveCharactersInDb(List<Character> characters);
    Single<EntityDb> getDataFromDb();

}

File: inKotlin/app/src/main/java/com/barrygithub/rickandmortyapp/data/remoteDatasource/Repository.kt
package com.barrygithub.rickandmortyapp.data.remoteDatasource


import com.barrygithub.rickandmortyapp.data.localDatasource.Character
import com.barrygithub.rickandmortyapp.data.localDatasource.EntityDb
import com.barrygithub.rickandmortyapp.data.localDatasource.MetaData
import io.reactivex.rxjava3.core.Single

/**
 * Project RickAndMortyApp
 * Created by Barry Zea H. on 14/04/23.
 * Copyright (c) Barry Zea H. All rights reserved.
 *
 **/
interface Repository {
    fun getDataFromGraphql(page:Int):Single<String>
    fun saveMetadataInDb(metadata: MetaData)
    fun saveCharactersInDb(characters: List<Character>)
    fun getDataFromDb(): Single<EntityDb>
}